core/controllers/improvements_test.py

```diff
@@ -397,44 +402,59 @@ def test_post_can_create_new_resolved_task_in_storage_with_utcnow(self):
             self.post_json(self.get_url(), {
                 'task_entries': [{
                     'entity_version': self.exp.version,
-                    'task_type': improvements_models.TASK_TYPE_HIGH_BOUNCE_RATE,
+                    'task_type': constants.TASK_TYPE_HIGH_BOUNCE_RATE,
                     'target_id': feconf.DEFAULT_INIT_STATE_NAME,
                     'issue_description': 'issue description',
-                    'status': improvements_models.TASK_STATUS_RESOLVED,
+                    'status': constants.TASK_STATUS_RESOLVED,
                 }]
             }, csrf_token=self.get_new_csrf_token())
 
         task_id = improvements_models.TaskEntryModel.generate_task_id(
-            improvements_models.TASK_ENTITY_TYPE_EXPLORATION,
+            constants.TASK_ENTITY_TYPE_EXPLORATION,
             self.exp.id,
             self.exp.version,
-            improvements_models.TASK_TYPE_HIGH_BOUNCE_RATE,
-            improvements_models.TASK_TARGET_TYPE_STATE,
+            constants.TASK_TYPE_HIGH_BOUNCE_RATE,
+            constants.TASK_TARGET_TYPE_STATE,
             feconf.DEFAULT_INIT_STATE_NAME)
         task_entry_model = improvements_models.TaskEntryModel.get_by_id(task_id)
 
         self.assertIsNotNone(task_entry_model)
         self.assertEqual(task_entry_model.id, task_id)
         self.assertEqual(
             task_entry_model.entity_type,
-            improvements_models.TASK_ENTITY_TYPE_EXPLORATION)
+            constants.TASK_ENTITY_TYPE_EXPLORATION)
         self.assertEqual(task_entry_model.entity_id, self.exp.id)
         self.assertEqual(task_entry_model.entity_version, self.exp.version)
         self.assertEqual(
             task_entry_model.task_type,
-            improvements_models.TASK_TYPE_HIGH_BOUNCE_RATE)
+            constants.TASK_TYPE_HIGH_BOUNCE_RATE)
         self.assertEqual(
             task_entry_model.target_type,
-            improvements_models.TASK_TARGET_TYPE_STATE)
+            constants.TASK_TARGET_TYPE_STATE)
         self.assertEqual(
             task_entry_model.target_id, feconf.DEFAULT_INIT_STATE_NAME)
         self.assertEqual(
             task_entry_model.issue_description, 'issue description')
         self.assertEqual(
-            task_entry_model.status, improvements_models.TASK_STATUS_RESOLVED)
+            task_entry_model.status, constants.TASK_STATUS_RESOLVED)
         self.assertEqual(task_entry_model.resolver_id, self.owner_id)
         self.assertEqual(task_entry_model.resolved_on, self.MOCK_DATE)
 
+    def test_to_dict_with_non_existing_resolver_id_raises_exception(
```

Comment:
Change "non_existing" to "invalid" -- the former sounds like it is missing/absent instead.

Maybe add a test for the "missing/absent" case (i.e. None) that ensures that there are no issues arising in that case.

Reply:
Done.